code/modules/mob/living/carbon/human/human.dm

```diff
@@ -539,13 +542,51 @@
 	popup.set_content(dat)
 	popup.open()
 
+/mob/living/carbon/human/proc/is_fat() //So we can check if someone is obese from genetics or from overeating
+	if((FAT in mutations) || (OBESITY in mutations))
+		return TRUE
 
 // called when something steps onto a human
-// this handles mulebots and vehicles
+// this handles mulebots, vehicles, and stomping
 /mob/living/carbon/human/Crossed(var/atom/movable/AM)
 	if(istype(AM, /obj/vehicle))
 		var/obj/vehicle/V = AM
 		V.RunOver(src)
+	if(ishuman(AM)) //Trample trample trample
+		var/mob/living/carbon/human/M = AM
+		if(world.time < M.laststomped + STOMP_COOLDOWN)
+			return
+		if(!(M.is_fat()) && M.a_intent == INTENT_HELP) //If they're fat or are on a non-help intent, then we can continue. Otherwise we don't trample them
+			return
+		if(M.buckled || M.throwing || M.m_intent == MOVE_INTENT_WALK) //If they're buckled into something, being thrown, or walking, they will not stomp
+			return
+		if(incapacitated() && !(M.incapacitated())) //If the victim is incapacitated and the assailant is not incapacitated. This also catches weird fringe cases where you might end up stomping yourself
+			M.trample(src)
+
+/mob/living/carbon/human/proc/trample(var/mob/living/carbon/human/H)
+	if(!(H.client)) //If they're SSD, we merely trip over them
+		H.visible_message("<span class = 'warning'>[src] trips on [H]!</span>", "<span class = 'warning'>[src] trips over you!</span>", "<span class = 'warning'>You hear someone stumbling!</span>")
+		Weaken(3)
+		return
+	var/D = TRAMPLING_DAMAGE
+	var/armour = H.run_armor_check()
+	if(is_fat()) //Being fat makes you tread with more force
+		D += 1
+	if(shoes)
+		var/obj/item/clothing/shoes/S = shoes
+		if(S.trampling_coefficient == 0) //No point treading on them if we're going to do 0 damage
+			return
+		D *= S.trampling_coefficient
+	else
+		D *= 0.25 //Standing on someone without shoes isn't that damaging
+	if(H.stat == DEAD)
+		H.visible_message("<span class = 'warning'>[src] treads upon [H]'s lifeless form.</span>", "<span class = 'warning'>[src] further violates your corpse.</span>", "<span class = 'warning'>You hear a fleshy thud.</span>")
+	else
+		H.visible_message("<span class = 'danger'>[src] stomps on [H]!</span>", "<span class = 'userdanger'>[src] squashes you under their feet!</span>", "<span class = 'warning'>You hear a fleshy thud.</span>")

```

Comment:
No, I mean something like `H.visible_message("You were trampled!" || if H.stat == "DEAD" "Your corpse was trampled!")`